tests/test_ragas.py

```diff
@@ -0,0 +1,161 @@
+# # SPDX-License-Identifier: Apache-2.0
+# Standard
+from pathlib import Path
+from unittest.mock import MagicMock, patch
+import unittest
+
+# Third Party
+from pandas import DataFrame
+from ragas.callbacks import ChainRun
+from ragas.dataset_schema import EvaluationDataset, EvaluationResult
+import pandas as pd
+
+# First Party
+from instructlab.eval.ragas import ModelConfig, RagasEvaluator, RunConfig, Sample
+
+
+class TestRagasEvaluator(unittest.TestCase):
+    @patch("instructlab.eval.ragas.get_openai_client")
+    def test_generate_answers_from_model(self, mock_get_openai_client):
+        # mock the OpenAI client to always return "london" for chat completions
+        mock_client = MagicMock()
+        mock_response = MagicMock()
+        mock_response.choices[0].message.content = "London"
+        mock_client.chat.completions.create.return_value = mock_response
+        mock_get_openai_client.return_value = mock_client
+
+        # get answers
+        questions = pd.DataFrame({"user_input": ["What is the capital of France?"]})
+        student_model = ModelConfig(
+            base_url="https://api.openai.com",
+            model_name="gpt-3.5-turbo",
+            api_key="test-api-key",
+        )
+        evaluator = RagasEvaluator()
+        result_df = evaluator._generate_answers_from_model(questions, student_model)
```

Comment:
@RobotSail shouldn't we hit a mock client here too.

Reply:
Instead of using gpt-3.5?

Reply:
@abhi1092 We do test it with a mock-client, GPT-3.5 was just the first model that came to mind as a fill-in.

Reply:
Updated this to use fake values to eliminate the possibility of it actually calling out to the real openai API.